subt/main.py

```diff
@@ -233,12 +236,17 @@ def go_safely(self, desired_direction):
         desired_angular_speed = 0.9 * safe_direction
         size = len(self.scan)
         dist = min_dist(self.scan[size//3:2*size//3])
-        if dist < 0.75:  # 2.0:
+        if dist < self.min_safe_dist:  # 2.0:
 #            desired_speed = self.max_speed * (1.2/2.0) * (dist - 0.4) / 1.6
-            desired_speed = self.max_speed * (dist - 0.4) / 0.35
+            desired_speed = self.max_speed * (dist - self.dangerous_dist) / (self.min_safe_dist - self.dangerous_dist)
         else:
             desired_speed = self.max_speed  # was 2.0
         desired_speed = desired_speed * (1.0 - 0.8 * min(self.max_angular_speed, abs(desired_angular_speed)) / self.max_angular_speed)
+        if self.min_speed:
```

Comment:
this is not the same as original code - in the old code the robot could also backup

Reply:
Sorry, i don't know what you mean. "the robot could also backup" ?

Reply:
The original robot can move backward (i.e. negative `desired_speed`) if the obstacle is too close. Now code prohibits that.

Reply:
Ok, it is fixed now. But i think there should be more sophisticated solution for the kloubaks. I am afraid the kloubak will be just jumping back and forth. I will test it tomorrow morning.

Reply:
Ok, it is fixed now. But i think there should be more sophisticated solution for the kloubaks. I am afraid the kloubak will be just jumping back and forth. I will test it tomorrow morning.